pkg/operator/ceph/object/spec.go

```diff
@@ -912,3 +922,93 @@ func renderProbe(cfg rgwProbeConfig) (string, error) {
 
 	return writer.String(), nil
 }
+
+func (c *clusterConfig) addDNSNamesToRGWServer() (string, error) {
+	if (c.store.Spec.Hosting == nil) || len(c.store.Spec.Hosting.DNSNames) <= 0 {
+		return "", nil
+	}
+	if !c.clusterInfo.CephVersion.IsAtLeastReef() {
+		return "", errors.New("rgw dns names are supported from ceph v18 onwards")
+	}
+
+	// add default RGW service name to dns names
+	dnsNames := c.store.Spec.Hosting.DNSNames
+	dnsNames = append(dnsNames, domainNameOfService(c.store))
+
+	// add custom endpoints from zone spec if exists
+	if c.store.Spec.Zone.Name != "" {
+		zone, err := c.context.RookClientset.CephV1().CephObjectZones(c.store.Namespace).Get(c.clusterInfo.Context, c.store.Spec.Zone.Name, metav1.GetOptions{})
+		if err != nil {
+			return "", err
+		}
+		dnsNames = append(dnsNames, zone.Spec.CustomEndpoints...)
+	}
```

Comment:
Are we missing store.Spec.Gateway.ExternalEndpoints here?

Reply:
The `external endpoints` are applicable for external rgw servers, there won't be an rgw pod running to configure spec this option. The external rgw server will be configured by the storage admin, Rook does not have much role there. Hence  external endpoints are not added